### PR TITLE
network: add user domain socket entry in bsd socket

### PR DESCRIPTION
**Pull request by pillip8282**
add user domain socket entry in bsd socket.
you can check whether it is user domain socket by the number of file descriptor

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [9ee84adecb68160dab90a3e12645aab7e6aaf6ca] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9ee84adecb68160dab90a3e12645aab7e6aaf6ca] - Code Rule Check (C++) OK.

File: os/net/netmgr/bsd_socket_api.c [hunk 1]
```diff
@@ -23,21 +23,25 @@
...
 #include <sys/types.h>
 #include <netinet/in.h>
 #include <net/if.h>
 #include <tinyara/lwnl/lwnl.h>
 #include "netstack.h"
+#include "uds.h"
 
-/*  Private */
+#define CONFIG_NUDS_DESCRIPTORS 0 // to be fixed: temporary
+#define AF_UNIX 1000 // to be fixed: temporary
```
Comment (junmin-kim) [inline os/net/netmgr/bsd_socket_api.c]: This line can be removed if #4126 is merged.
How about reviewing #4126 first?

File: os/net/netmgr/bsd_socket_api.c [hunk 2]
```diff
@@ -49,84 +53,153 @@ int _create_netlink(int type, int protocol)
 	return fd;
 }
 
+sock_type _get_socktype(int fd)
+{
+	if (fd < CONFIG_NFILE_DESCRIPTORS) {
+		return LWNL;
+	} else if (fd < CONFIG_NUDS_DESCRIPTORS) {
```
Comment (junmin-kim) [inline os/net/netmgr/bsd_socket_api.c]:
```suggestion
	} else if (fd < CONFIG_NFILE_DESCRIPTOR + CONFIG_NUDS_DESCRIPTORS) {
```

File: os/net/netmgr/bsd_socket_api.c [hunk 3]
```diff
@@ -49,84 +53,153 @@ int _create_netlink(int type, int protocol)
 	return fd;
 }
 
+sock_type _get_socktype(int fd)
```
Comment (junmin-kim) [inline os/net/netmgr/bsd_socket_api.c]: Where is the definition of sock_type?

File: os/net/netmgr/bsd_socket_api.c (hunk 1 above)
Comment (pillip8282) [reply os/net/netmgr/bsd_socket_api.c]: ok

File: os/net/netmgr/bsd_socket_api.c (hunk 2 above)
Comment (pillip8282) [reply os/net/netmgr/bsd_socket_api.c]: ok, I'll change it after your PR is merged

File: os/net/netmgr/bsd_socket_api.c (hunk 3 above)
Comment (pillip8282) [reply os/net/netmgr/bsd_socket_api.c]: I missed uploading. I'll add it

**Comment by seinfra** [comment]
Target : [5e35865c3e72961d6228824900af53d24aaa6278] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5e35865c3e72961d6228824900af53d24aaa6278] - Code Rule Check OK.

File: os/net/netmgr/bsd_socket_api.c (hunk 2 above)
Comment (pillip8282) [reply os/net/netmgr/bsd_socket_api.c]: Thanks, I'll modify it

**Comment by seinfra** [comment]
Target : [7ff2599b11fb7c8018a204fff045fc52c6bd4e2c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7ff2599b11fb7c8018a204fff045fc52c6bd4e2c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f7fed773d096e851b65c711763f4458de7fc202c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f7fed773d096e851b65c711763f4458de7fc202c] - Code Rule Check (C++) OK.

File: os/net/netmgr/bsd_socket_api.c [hunk 4]
```diff
@@ -23,21 +23,25 @@
...
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <netinet/in.h>
 #include <net/if.h>
 #include <tinyara/lwnl/lwnl.h>
 #include "netstack.h"
+#include "uds.h"
 
-/*  Private */
+#define CONFIG_NUDS_DESCRIPTORS 0 // to be fixed: temporary
```
Comment (junmin-kim) [inline os/net/netmgr/bsd_socket_api.c]:
```suggestion
#ifndef CONFIG_NUDS_DESCRIPTORS
#define CONFIG_NUDS_DESCRIPTORS 0 // to be fixed: temporary
#endif
```
Would you change it tmporarily?
I'll remove this line after adding CONFIG_NUDS_DESCRIPTORS in Kconfig.

File: os/net/netmgr/bsd_socket_api.c (hunk 1 above)
Comment (junmin-kim) [reply os/net/netmgr/bsd_socket_api.c]: Now this line can be removed.

**Comment by seinfra** [comment]
Target : [1ea50d0c60abc99dfbc76b07d0a5091fade844ba] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1ea50d0c60abc99dfbc76b07d0a5091fade844ba] - Code Rule Check (C++) OK.

**Pull request merged**